sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -496,6 +496,16 @@ class kernel_bundle_impl {
 
   bool isInterop() const { return MIsInterop; }
 
+  void filterImages(const kernel_id &KernelId) {
+    std::vector<device_image_plain> FilteredDeviceImages;
+    std::copy_if(MDeviceImages.begin(), MDeviceImages.end(),
+                 std::back_inserter(FilteredDeviceImages),
+                 [&KernelId](const device_image_plain &Img) {
+                   return Img.has_kernel(KernelId);
+                 });
+    MDeviceImages = FilteredDeviceImages;
```

Comment:
Can the following be faster than the current version?
```
for (auto It = MDeviceImages.begin(); It != MDeviceImages.end();) {
  device_image_plain &Img = *It;
  if (!Img.has_kernel(KernelId)) {
    std::swap(Img, *MDeviceImages.rbegin());
    MDeviceImages.pop_back();
  } else
    ++It;
}
```

Here, there's no copying, but, probably, it's a worse practicing. Though, `swap` might do the copy stuff.
@romanovvlad , any thoughts?

Reply:
It should be faster I believe, at least we avoid dynamic allocations for the temp vector.

But this alternative may be simpler for reading:
```
size_t InsertPos = 0;
for(size_t I = 0; I < MDeviceImages.size(); ++I) {
  if(!MDeviceImages[I].has_kernel(KernelId))
     continue;

  std::swap(MDeviceImages[I], MDeviceImages[InsertPos]);
  InsertPos++;
}
MDeviceImages.resize(InsertPos);
```